hooks/VDM.js

```diff
@@ -268,19 +304,304 @@ module.exports = function (input, session) {
+        value: utils.transform(data.winddir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.windgustdir) {
+    contextPrefix = 'meteo.'
+    if (data.windgustdir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.wind.gustDirection`,
+        value: utils.transform(data.windgustdir, 'deg', 'rad'),
+      })
+    }
+  }
+
+  if (data.airtemp) {
+    contextPrefix = 'meteo.'
+    if (data.airtemp > -601 && data.airtemp < 601) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.temperature`,
+        value: utils.transform((data.airtemp / 10), 'c', 'k'),
+      })
+    }
+  }
+
+  if (data.relhumid) {
+    contextPrefix = 'meteo.'
+    if (data.relhumid < 101) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.relativeHumidity`,
+        value: data.relhumid,
+      })
+    }
+  }
+
+  if (data.dewpoint) {
+    contextPrefix = 'meteo.'
+    if (data.dewpoint > -201 && data.dewpoint < 501) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.dewPoint`,
+        value: utils.transform((data.dewpoint / 10), 'c', 'k'),
+      })
+    }
+  }
+
+  if (data.airpress) {
+    contextPrefix = 'meteo.'
+    if (data.airpress < 403) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.pressure`,
+        value: (data.airpress + 799) * 100,
+      })
+    }
+  }
+
+  if (data.airpressten) {
+    contextPrefix = 'meteo.'
+    if (data.airpressten < 3) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.pressureTendency`,
+        value: statusTable[data.airpressten],
+      })
+    }
+  }
+
+  if (data.horvisib) {
+    contextPrefix = 'meteo.'
+    if (data.horvisib < 127) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.air.horizontalVisibility`,
+        value: utils.transform((data.horvisib / 10), 'nm', 'm'),
+      })
+    }
+  }
+
+  if (data.waterlevel) {
+    contextPrefix = 'meteo.'
+    if (data.waterlevel < 4001) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.level`,
+        value: (data.waterlevel / 100) - 10,
+      })
+    }
+  }
+
+  if (data.waterlevelten) {
+    contextPrefix = 'meteo.'
+    if (data.waterlevelten < 3) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.water.levelTrend`,
+        value: statusTable[data.waterlevelten],
+      })
+    }
+  }
+
+  if (data.surfcurrspd) {
+    contextPrefix = 'meteo.'
+    if (data.surfcurrspd < 252) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.surfaceCurrent.speed`,
+        value: utils.transform((data.surfcurrspd / 10), 'knots', 'ms'),
+      })
+    }
+  }
+
+  if (data.surfcurrdir) {
+    contextPrefix = 'meteo.'
+    if (data.surfcurrdir < 360) {
+      values.push({
+        path: `environment.observations.${meteoLocation}.surfaceCurrent.direction`,
```

Comment:
Should this not be environment.observations.${meteoLocation}.water.surfaceCurrent.direction?